omniport/core/open_auth/views/retrieve_data.py

```diff
@@ -86,11 +93,17 @@ def get(self, request):
                 response_data[model_name] = dict()
 
         if 'person.roles' in app_data_points:
-            response_data['person']['roles'] = get_roles(person)
+            try:
+                response_data['person']['roles'] = get_roles(person)
```

Comment:
Why are these kinds of `KeyError` exceptions to be handled? If `person` key does not exist then how can the roles be defined?
I am firmly against the structure of response API being different for different use cases. Even if there is some use case where `person` is not required but `roles` and `display_picture` are (which I find entirely absurd), then also the `person` key must be there with `roles` and `display_picture` keys as children keys.

Reply:
I've changed that part. But regarding the roles and display pictures being sent and the person not, that actually depends on the scope the developer has selected at the time of app registration.

Reply:
Yeah, but in the scopes also `roles` and `display_picture` are child keys for `person`. The `person` key should be defined according to `scopes` also. I think PR is good after the latest change.
Just for the clarification on the use case pointed out by @ayushjn20, there isn't any option to select the parent keys, only child keys are selected, so the situation is that developer didn't ask for other details of `person` key like `full_name` and `short_name`.